cura/Settings/MachineManager.py

```diff
@@ -1496,3 +783,268 @@ def setSettingForAllExtruders(self, setting_name: str, property_name: str, prope
         for key, extruder in self._global_container_stack.extruders.items():
             container = extruder.userChanges
             container.setProperty(setting_name, property_name, property_value)
+
+    #
+    # New
+    #
+
+    # We not fetch it from _current_root_material_id, but later we can get it from somewhere else
+    @pyqtProperty("QVariantList", notify = rootMaterialChanged)
+    def currentExtruderPositions(self):
+        return sorted(list(self._current_root_material_id.keys()))
```

Comment:
Why are you depending on the root material IDs to get the extruder positions? Use `global_stack.extruders` or something?

Reply:
Or maybe even `range(len(global_stack.extruders))` or something of the like.

Reply:
I've used `global_stack.extruders` directly now: f40e9bffa9aeee2e312fbe4cbe5cab7bfa8189f6